src/interfaces.d.ts

```diff
@@ -214,7 +214,8 @@ export interface WidgetOptions<S extends WidgetState, P extends WidgetProperties
 }
 
 export interface WidgetProperties {
-	[key: string]: any;
+	id?: string;
```

Comment:
This are properties that are used in createWidgetBase so should be on the default properties interface.